server/controllers/userController.js

```diff
@@ -1,25 +1,22 @@
-import * as db from '../models/swatchModel';
+import { db } from '../models/swatchModel.js';
 import bcrypt from 'bcrypt';
 
-SALT_WORK_FACTOR = 10;
+const SALT_WORK_FACTOR = 10;
 
 export const userController = {};
 
-userController.createUser = (req, res, next) => {
+userController.createUser = async (req, res, next) => {
   const { username, email, password, key } = req.body;
 
   bcrypt.hash(password, SALT_WORK_FACTOR)
     .then(hash => {
       const query = `
         INSERT INTO users
-          (username, email, password, key)
-        SELECT
-          $1, $2, $3, $4
-        WHERE NOT EXISTS (
-          SELECT email
-          FROM users
-          WHERE email = $2
-        )
+          (username, email, password, validateKey)
+        VALUES
+          ($1, $2, $3, $4)
+        ON CONFLICT (email) DO NOTHING
+        RETURNING *;
```

Comment:
Do you need _everything_ to be returned? Perhaps just `id`, or `username` would be sufficient

Reply:
fixed. will be included in next commit.